taskipy/pyproject.py

```diff
@@ -0,0 +1,65 @@
+from pathlib import Path
+from typing import Any, MutableMapping
+
+import toml
+
+from taskipy.exceptions import (
+    MalformedPyProjectError,
+    MissingPyProjectFileError,
+    MissingTaskipyTasksSectionError
+)
+
+
+class PyProject:
+    def __init__(self, file_name: str = "pyproject.toml", directory: Path = Path.cwd()):
+        """
+        Initializes PyProject by setting up the path to and loading in
+        the pyproject.toml file in the current directory.
+
+        Args:
+            file_name (str, optional): The name of the pyproject file. Defaults to 'pyproject.toml'.
+            directory (Path, optional): The directory that the pyproject file is in. Defaults to Path.cwd().
+        """
+        self.path = Path(directory / file_name).resolve()
+        self.items = self.__load_toml_file()
```

Comment:
👍 as a side note, there may be a much better name than `items`, just couldn't think of one. At that point, you may not even need `path` since it's only used by `__load_toml_file()`. It could just be passed in there to make it more explicit that is the toml file that is being loaded and not have it stored if it'll be private anyway. 

I think the thinking there behind saving it was that the path could be useful for debugging purposes such as maybe passing it into the `MissingPyProjectFileError` so the error message could be more helpful by showing what path was being looking at to find the pyproject file. Your call, let me know what you think.